redbot/cogs/cleanup/cleanup.py

```diff
@@ -541,3 +541,25 @@ def check(m):
             await mass_purge(to_delete, channel)
         else:
             await slow_deletion(to_delete)
+
+    @cleanup.command(name="spam")
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def cleanup_spam(self, ctx: commands.Context, number: int = 50):
+        """Deletes duplicate messages in the channel from the last X messages. Default 50."""
+        msgs = []
+        spam = []
+        async for msg in ctx.channel.history(limit=number + 1):
```

Comment:
Instead of using `number + 1` for the command to check 50 messages excluding command message, it might be better to use `before` kwarg. This will also make sure that the bot will not include some message that happens to be sent right after the command message but before we the bot actually requests the channel's history.
```suggestion
        async for msg in ctx.channel.history(limit=number, before=ctx.message):
```